pvanalytics/tests/quality/test_irradiance.py

```diff
@@ -411,3 +411,199 @@ def test_calculate_dni_component(generate_RMIS_irradiance_series):
+    poa_global_limit_bool_flag, poa_global_limit_int_flag = \
+        irradiance.check_poa_global_limits_lorenz(poa_global, solar_zenith,
+                                                  aoi, dni_extra)
+
+    assert_series_equal(expected_int_flag,
+                        poa_global_limit_int_flag,
+                        check_names=False)
+
+    assert_series_equal(poa_global_limit_bool_flag,
+                        expected_bool_flag,
+                        check_names=False)
+
+
+def test__upper_poa_global_limit_lorenz(lorenz_test_data):
+    """Testing upper poa global limit defined by Lorenz et al."""
+
+    data = lorenz_test_data
+
+    # Expected upper limit
+    expected_upper_limit = data['upper_limit_poa']
+
+    # Setting up inputs
+    dni_extra = 1367
+    solar_zenith = data['solar_zenith']
+    aoi = data['aoi']
+
+    # Testing upper limit
+    test_upper_limit = irradiance._upper_poa_global_limit_lorenz(aoi,
+                                                                 solar_zenith,
+                                                                 dni_extra)
+
+    assert_series_equal(expected_upper_limit,
+                        test_upper_limit,
+                        check_names=False)
+
+
+def test__upper_ghi_limit_lorenz_flag2(lorenz_test_data):
+    """Testing upper ghi limit for flag 2 defined by Lorenz et al."""
+
+    data = lorenz_test_data
+
+    # Expected upper limit
+    expected_upper_limit = data['upper_limit_flag2_ghi']
+
+    # Setting up inputs
+    dni_extra = 1367
+    solar_zenith = data['solar_zenith']
+
+    # Testing upper limit
+    test_upper_limit = irradiance._upper_ghi_limit_lorenz_flag2(solar_zenith,
+                                                                dni_extra)
+
+    assert_series_equal(expected_upper_limit,
+                        test_upper_limit,
+                        check_names=False)
+
+
+def test__upper_ghi_limit_lorenz_flag3(lorenz_test_data):
+    """Testing upper ghi limit for flag 3 defined by Lorenz et al."""
+
+    data = lorenz_test_data
+
+    # Expected upper limit
+    expected_upper_limit = data['upper_limit_flag3_ghi']
+
+    # Setting up inputs
+    dni_extra = 1367
+    solar_zenith = data['solar_zenith']
+
+    # Testing upper limit
+    test_upper_limit = irradiance._upper_ghi_limit_lorenz_flag3(solar_zenith,
+                                                                dni_extra)
+
+    assert_series_equal(expected_upper_limit,
+                        test_upper_limit,
+                        check_names=False)
+
+
+def test__lower_limit_lorenz(lorenz_test_data):
+    """Testing lower ghi limit defined by Lorenz et al."""
+
+    data = lorenz_test_data
+
+    # Expected upper limit
+    expected_upper_limit = data['lower_limit_ghi']
+
+    # Setting up inputs
+    dni_extra = 1367
+    solar_zenith = data['solar_zenith']
+
+    # Testing upper limit
+    test_upper_limit = irradiance._lower_limit_lorenz(solar_zenith,
```

Comment:
Several comments and variable names here should be `lower_limit`

Reply:
Ah - that's a bit embarrassing. Thanks for pointing!